File: DriverTemplates/TemplateSources/src/ASCOM Driver Template (VB)/DeviceDome.vb
' All lines from line 1 to the device interface implementation region will be discarded by the project wizard when the template is used
' Required code must lie within the device implementation region
' The //ENDOFINSERTEDFILE tag must be the last but one line in this file

Imports ASCOM.DeviceInterface
Imports ASCOM
Imports ASCOM.Utilities

Class DeviceDome
    Implements IDomeV2
    Private m_util As New Util()
    Private TL As New TraceLogger()

#Region "IDome Implementation"
    Public Sub AbortSlew() Implements IDomeV2.AbortSlew
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public ReadOnly Property Altitude() As Double Implements IDomeV2.Altitude
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property AtHome() As Boolean Implements IDomeV2.AtHome
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property AtPark() As Boolean Implements IDomeV2.AtPark
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property Azimuth() As Double Implements IDomeV2.Azimuth
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanFindHome() As Boolean Implements IDomeV2.CanFindHome
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanPark() As Boolean Implements IDomeV2.CanPark
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanSetAltitude() As Boolean Implements IDomeV2.CanSetAltitude
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanSetAzimuth() As Boolean Implements IDomeV2.CanSetAzimuth
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanSetPark() As Boolean Implements IDomeV2.CanSetPark
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanSetShutter() As Boolean Implements IDomeV2.CanSetShutter
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanSlave() As Boolean Implements IDomeV2.CanSlave
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property CanSyncAzimuth() As Boolean Implements IDomeV2.CanSyncAzimuth
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Sub CloseShutter() Implements IDomeV2.CloseShutter
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public Sub FindHome() Implements IDomeV2.FindHome
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public Sub OpenShutter() Implements IDomeV2.OpenShutter
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public Sub Park() Implements IDomeV2.Park
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public Sub SetPark() Implements IDomeV2.SetPark
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public ReadOnly Property ShutterStatus() As ShutterState Implements IDomeV2.ShutterStatus
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Property Slaved() As Boolean Implements IDomeV2.Slaved
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
        Set(value As Boolean)
            Throw New ASCOM.PropertyNotImplementedException()
        End Set
    End Property

    Public Sub SlewToAltitude(Altitude As Double) Implements IDomeV2.SlewToAltitude
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public Sub SlewToAzimuth(Azimuth As Double) Implements IDomeV2.SlewToAzimuth
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public ReadOnly Property Slewing() As Boolean Implements IDomeV2.Slewing
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Sub SyncToAzimuth(Azimuth As Double) Implements IDomeV2.SyncToAzimuth
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

#End Region

    '//ENDOFINSERTEDFILE
End Class

File: DriverTemplates/TemplateSources/src/ASCOM Driver Template (CS)/DeviceDome.cs
// All lines from line 1 to the device interface implementation region will be discarded by the project wizard when the template is used
// Required code must lie within the device implementation region
// The //ENDOFINSERTEDFILE tag must be the last but one line in this file

using ASCOM.DeviceInterface;
using System;
using ASCOM;
using ASCOM.Utilities;

class DeviceDome
{
    Util util = new Util(); TraceLogger tl = new TraceLogger();

    #region IDome Implementation
    public void AbortSlew()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public double Altitude
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool AtHome
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool AtPark
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public double Azimuth
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanFindHome
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanPark
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanSetAltitude
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanSetAzimuth
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanSetPark
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanSetShutter
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanSlave
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool CanSyncAzimuth
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public void CloseShutter()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public void FindHome()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public void OpenShutter()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public void Park()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public void SetPark()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public ShutterState ShutterStatus
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool Slaved
    {
        get
        {
            throw new ASCOM.PropertyNotImplementedException();
        }
        set
        {
            throw new ASCOM.PropertyNotImplementedException();
        }
    }

    public void SlewToAltitude(double Altitude)
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public void SlewToAzimuth(double Azimuth)
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public bool Slewing
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public void SyncToAzimuth(double Azimuth)
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    #endregion

    //ENDOFINSERTEDFILE
}